src/pykeen/training/accelerated_training_loop.py

```diff
@@ -0,0 +1,454 @@
+# -*- coding: utf-8 -*-
+
+"""Accelerated Training loop for KGE models using HF Accelerate"""
+
+import gc
+import logging
+import pathlib
+import time
+from abc import ABC
+from typing import Any, List, Mapping, Optional, Type, Union
+
+import torch
+from accelerate import Accelerator
+from torch.optim.optimizer import Optimizer
+from torch.utils.data import DataLoader
+from tqdm.autonotebook import tqdm, trange
+
+from .training_loop import NonFiniteLossError, SubBatchingNotSupportedError, TrainingLoop, _get_optimizer_kwargs
+from ..losses import Loss, has_mr_loss
+from ..models import Model, RGCN
+from ..stoppers import Stopper
+from ..trackers import ResultTracker
+from ..training.schlichtkrull_sampler import GraphSampler
+from ..triples import Instances
+from ..utils import format_relative_comparison, get_batchnorm_modules
+
+__all__ = [
+    'AcceleratedTrainingLoop',
+]
+
+logger = logging.getLogger(__name__)
+
+
+class AlternateAcceleratedTrainingLoop(TrainingLoop, ABC):
+    """A training loop with HF accelerate."""
+
+    def __init__(
+        self,
+        model: Model,
+        optimizer: Optional[Optimizer] = None,
+        **kwargs,
+    ) -> None:
+        """Initialize the training loop.
+
+        :param model: The model to train
+        :param optimizer: The optimizer to use while training the model
+        :param automatic_memory_optimization: bool
+            Whether to automatically optimize the sub-batch size during
+            training and batch size during evaluation with regards to the hardware at hand.
+        """
+        super().__init__(model=model, optimizer=optimizer, **kwargs)
```

Comment:
Since, `model` and `optimizer` are not used in this class' `__init__`, we can move them into the `kwargs`, i.e.

```python
def __init__(self, *, **kwargs) -> None:
    super().__init__(**kwargs)
    self.accelerator = Accelerator()
```

Reply:
excellent idea